examples/apps/ncp/main.c

```diff
@@ -97,3 +97,24 @@ int main(int argc, char *argv[])
 
     return 0;
 }
+
+
+/*
+ * Provide, if required an "otPlatLog()" function
+ */
+
+#if ((OPENTHREAD_CONFIG_LOG_OUTPUT == OPENTHREAD_CONFIG_LOG_OUTPUT_NONE) ||  (OPENTHREAD_CONFIG_LOG_OUTPUT == OPENTHREAD_CONFIG_LOG_OUTPUT_APP_STREAM))
```

Comment:
That does not help here.

The Platform library (and thus  the platform provided/required otPlatLog() function) is is *common* to both the ncp and the cli - it does not know which function to call/invoke.

That's why we have these outputs,

NONE = the bit bucket no output

DEBUG_UART - the common debug uart scheme

APP_STREAM - as defined by the application layer (ie: CLI or NCP, or in the future a SENSOR MODULE, or THERMOSTAT module) - maybe this should be called APP_PROVIDED

PLATFORM_PROVIDED - is for situations like Nordics RTT - we at TI have a diffferent solution that works with our tool chain and thus it is really "platform level"

POSIX - is really sort a special case of platform provided